### ISSUE TITLE
error with tmp_dir in container_test (TALON)

### ISSUE THREAD

**Issue by slsevilla** [RBL-NCI/RBL_RBL3#3]
When running talon a talon_tmp directory is created wherever the script is written from. It seems that when running talon within the container, it cannot find this dir, and is erring that the bam file cannot be found. This error does not occur when using "pip install".

I've reviewed the doc's for TALON and there is no tmp_dir flag as there is in other functions. Someone else already created an issue around this, and I've added to it, explaining the importance for this. In the meantime, I've updated the pipeline to use pip install, and we can work to redirect the container version or perhaps add the tmp_dir flag.

**Comment by kopardev** [comment]
Edited the talon.py to accept a required tmp_dir argument like this: 
```
# talon
[ 2021-03-24 14:11:37 ] Started TALON run
usage: talon [-h] [--f CONFIG_FILE] [--db FILE,] [--build STRING,]
             [--threads THREADS] [--cov MIN_COVERAGE] --tmp_dir TMP_DIR
             [--identity MIN_IDENTITY] [--o OUTPREFIX]
talon: error: the following arguments are required: --tmp_dir
```
Added this version to the new [docker](https://hub.docker.com/layers/142799753/nciccbr/ccbr_talon_v5.0/2.0/images/sha256-0c804c952f07ff9f75933b3e4fe64cb92678792fae149a3c7b5e5f6e73053f70?context=explore)

**Comment by kopardev** [comment]
`container_test` branch has the latest container with `--tmp_dir` as a required argument for talon

**Comment by slsevilla** [comment]
testing with latest pull, had new error:

```
rule talon_annotation:
    input: /data/sevillas2/rbl3/03_talon/SIRV.db, /data/sevillas2/rbl3/03_talon/talon_config.csv, /data/sevillas2/rbl3/04_sam_labeled/barcode01_labeled.sam, /data/sevillas2/rbl3/04_sam_labeled/barcode02_labeled.sam, /data/sevillas2/rbl3/04_sam_labeled/barcode03_labeled.sam
    output: /data/sevillas2/rbl3/05_annotation/SIRV_talon_read_annot.tsv, /data/sevillas2/rbl3/05_annotation/SIRV_QC.log
    jobid: 0


        talon             --f /data/sevillas2/rbl3/03_talon/talon_config.csv             --db /data/sevillas2/rbl3/03_talon/SIRV.db             --build SIRV             --threads 5             --tmp_dir /data/sevillas2/rbl3/tmp_annotations             --o /data/sevillas2/rbl3/05_annotation/SIRV
        
Activating singularity image /gpfs/gsfs4/users/CCBR/projects/rbl3/.snakemake/singularity/fc13b4226faaf1a7f9ae841924754fce.simg
[ 2021-03-24 23:10:19 ] Started TALON run
[ 2021-03-24 23:19:04 ] Merged input SAM/BAM files
"bamToBed" does not appear to be installed or on the path, so this method is disabled.  Please install a more recent version of BEDTools and re-import to use this method.
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/talon/process_sams.py", line 78, in partition_reads
    all_reads = pybedtools.BedTool(merged_bam).bam_to_bed()
  File "/usr/local/lib/python3.7/dist-packages/pybedtools/bedtool.py", line 917, in decorated
    result = method(self, *args, **kwargs)
  File "/usr/local/lib/python3.7/dist-packages/pybedtools/bedtool.py", line 240, in not_implemented_func
    raise NotImplementedError(help_str)
NotImplementedError: "bamToBed" does not appear to be installed or on the path, so this method is disabled.  Please install a more recent version of BEDTools and re-import to use this method.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/bin/talon", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.7/dist-packages/talon/talon.py", line 2431, in main
    read_groups, intervals, header_file = procsams.partition_reads(sam_files, datasets, tmp_dir = options.tmp_dir)
  File "/usr/local/lib/python3.7/dist-packages/talon/process_sams.py", line 81, in partition_reads
    raise RuntimeError("Problem opening sam file %s" % (merged_bam))
RuntimeError: Problem opening sam file /data/sevillas2/rbl3/tmp_annotationsmerged.bam

```

It appears there is an issue with bedtools installed in the container. I've moved the test data I've used to this dir if you want to test to reproduce the error: /scratch/sevillas2/rbl3/

**Comment by kopardev** [comment]
Fixed... I overwrote the docker...can you try again?

**Comment by slsevilla** [comment]
issue resolved with commit 855d95c64d51f1eaea5992561dab7897d05a383b and latest TALON docker
